ERCS/erc-7390.md

```diff
@@ -0,0 +1,366 @@
+
+*Emits `Canceled` event if the cancelation was successful.*
+
+#### `updatePremium`
+
+```solidity
+function updatePremium(uint256 id, uint256 amount) external;
+```
+
+Allows the seller to update the premium that option buyer will need to provide for buying the options. Note that the `amount` will be for the whole underlying amount, not only for the options that might still be available for purchase.
+
+*Emits `PremiumUpdated` event when the function call was handled successfully.*
+
+#### `issuance`
+
+```solidity
+function issuance(uint256 id) external view returns (OptionIssuance memory);
+```
+
+Returns all the key information for the option issuance with the given `id`.
+
+### Events
+
+#### `Created`
+
+```solidity
+event Created(uint256 id);
+```
+
+Emitted when the writer has provided option issuance data successfully (and locked down the collateral to the contract). The given `id` identifies the particualr option issuance.
+
+#### `Bought`
+
+```solidity
+event Bought(uint256 indexed id, uint256 amount, address indexed buyer);
+```
+
+Emitted when options have been bought. Provides information about the option issuance `id`, the address of `buyer` and the `amount` of options bought.
+
+#### `Exercised`
+
+```solidity
+event Exercised(uint256 indexed id, uint256 amount);
+```
+
+Emitted when the option has been exercised from the option issuance with given `id` and the given `amount`.
+
+#### `Expired`
+
+```solidity
+event Expired(uint256 indexed id);
+```
+
+Emitted when the seller of the option issuance with `id` has retrieved the un-exercised collateral.
+
+#### `Canceled`
+
+```solidity
+event Canceled(uint256 indexed id);
+```
+
+Emitted when the option issuance with given `id` has been cancelled by the seller.
+
+#### `PremiumUpdated`
+
+```solidity
+event PremiumUpdated(uint256 indexed id, uint256 amount);
+```
+
+Emitted when seller updates the premium to `amount` for option issuance with given `id`. Note that the updated premium is for the total issuance.
+
+### Concrete Examples
+
+#### Call Option
+
+Let's say Bob sells an **call** option that Alice wants to buy.\
+He gives the right to Alice to buy **8 LINK** tokens at **25 USDC** each between **14th of July 2023** and **16th of July 2023**.\
+For such a contract, he asks Alice to give him **10 DAI** as a premium.\
+
+To create the contract, he will give the following parameters:
+
+- `side`: **Call**
+- `underlyingToken`: **0x514910771AF9Ca656af840dff83E8264EcF986CA** *(LINK's address)*
+- `amount`: **8000000000000000000** *(8 \* 10^(LINK's decimals))*
+- `strikeToken`: **0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48** *(USDC's address)*
+- `strike`: **200000000** *(8 * 25 \* 10^(USDC's decimals))*
+- `premiumToken`: **0x6B175474E89094C44Da98b954EedeAC495271d0F** *(DAI's address)*
```

Comment:
I'd recommend using fictitious addresses for your examples, though that isn't strictly required.